app/src/main/java/com/tristanwiley/chatse/chat/adapters/RoomAdapter.kt

```diff
@@ -25,14 +25,30 @@ import org.jetbrains.anko.doAsync
  */
 class RoomAdapter(val site: String, val list: MutableList<Room>, val context: Context) : RecyclerView.Adapter<RoomAdapter.ListRowHolder>() {
 
+
+    interface OnItemClickListener {
+
+        fun onItemClick(chatRoom: ChatRoom)
+    }
+
+
+
+    private val roomClickListener:OnItemClickListener
+
+    init {
+        roomClickListener = context as OnItemClickListener
```

Comment:
I'm not sure I like this. What if you pass a context that doesn't implement `OnItemClickListener`? For example: `getApplicationContext()`. @TimCastelijns I'd like your opinion on this.

Reply:
what do you suggest we change here?

Reply:
Perhaps we could explicitly require the listener interface on the constructor. But I'll wait Tim's opinion on this matter.

Reply:
Yes definitely avoid casting and pass the listener in the constructor.